sql/core/src/test/resources/sql-tests/inputs/higher-order-functions.sql

```diff
@@ -11,10 +11,10 @@ select upper(x -> x) as v;
 select transform(zs, z -> z) as v from nested;
 
 -- Transform an array
-select transform(ys, y -> y * y) as v from nested;
+select transform(ys, y1 -> y1 * y1) as v from nested;
```

Comment:
If `y` cannot be used here, this PR introduce a behavior change. At least, it should be under a flag.

Reply:
I just found out that all non reversed keywords can not be used in high order functions，is this a bug？？also cc @cloud-fan

Reply:
Do you mean enable the this whole feature or just y only？under ansi.enabled？

Reply:
Considering this as a shortage of high order functions, I'd like to fix it in #26366

Reply:
since that pr is merged i will revert these change